src/PowerShell/Microsoft.WinGet.Configuration.Engine/Commands/ConfigurationCommand.cs

```diff
@@ -377,18 +397,84 @@ private IConfigurationSetProcessorFactory CreateFactory(OpenConfigurationParamet
             return factory;
         }
 
+        private async Task<IConfigurationSetProcessorFactory> CreateDSCv3ProcessorFactory(OpenConfigurationParameters openParams)
+        {
+            var factory = new DSCv3ConfigurationSetProcessorFactory();
+
+            var factoryMap = factory.As<IDictionary<string, string>>();
+            if (!string.IsNullOrEmpty(openParams.ProcessorPath))
+            {
+                factoryMap.Add(DSCv3FactoryMapKeyDscExecutablePath, openParams.ProcessorPath);
+            }
+            else
+            {
+                string? foundProcessorPath = null;
+                if (!factoryMap.TryGetValue(DSCv3FactoryMapKeyFoundDscExecutablePath, out foundProcessorPath) ||
+                    string.IsNullOrEmpty(foundProcessorPath))
+                {
+                    await this.InstallDSCv3Package(openParams);
+                }
+            }
+
+            return factory;
+        }
+
+        private async Task InstallDSCv3Package(OpenConfigurationParameters openParams)
+        {
+            this.Write(StreamType.Information, Resources.ConfigurationInstallDscPackage);
+
+            InitialSessionState initialSessionState = InitialSessionState.CreateDefault();
+            initialSessionState.ExecutionPolicy = openParams.ExecutionPolicy;
+            Runspace runspace = RunspaceFactory.CreateRunspace(initialSessionState);
+            runspace.Open();
+            PowerShell installDSCv3 = PowerShell.Create(runspace).AddScript(
+                $@"
+                if (-not (Get-Module -ListAvailable -Name {WinGetClientModule}))
+                {{
+                    Install-Module -Name {WinGetClientModule} -Confirm:$False -Force
+                }}
+                Install-WingetPackage -Id {DSCv3PackageId} -Source msstore");
+
+            await installDSCv3.InvokeAsync();
+
+            if (installDSCv3.HadErrors)

```

Comment:
`Install-WinGetPackage` does not produce an error on failed install.  The result object must be inspected.

I'm not sure of the "correct" PowerShell design, but maybe one of:
1. A new cmdlet like `Validate-WinGetPackageInstall` that throws if the install failed
2. A new parameter like `-ThrowOnFailure`
3. A breaking change to throw on failure (maybe with a parameter to not)

Reply:
I can change the script to inspect the result object.